From 9beae9adb34795ca8f67f7a21d0b6ac3028c32b7 Mon Sep 17 00:00:00 2001
From: Nicolin Chen <Guangyu.Chen@freescale.com>
Date: Tue, 24 Jun 2014 14:44:18 +0800
Subject: [PATCH 0568/1691] MLK-11429-17: ASoC: fsl_spdif: Complete the
 volatile register list

commit 9beae9adb34795ca8f67f7a21d0b6ac3028c32b7 from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

cherry-pick below patch from imx_3.14.y
ENGR00320241 ASoC: fsl_spdif: Complete the volatile register list

Not only SIS but also other read-only or write-only reigsters should be marked
as volatile register so as not to let regcache cache them. So this patch just
adds those missing registers.

Signed-off-by: Nicolin Chen <Guangyu.Chen@freescale.com>
(cherry picked from commit 56ad235ba23274fc05422ad4b13153d1c56801c4)
(cherry picked from commit 43825f0150132dd51136b10d1aa83646be7b780a)
---
 sound/soc/fsl/fsl_spdif.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/sound/soc/fsl/fsl_spdif.c b/sound/soc/fsl/fsl_spdif.c
index cda3341..b4c18a4 100644
--- a/sound/soc/fsl/fsl_spdif.c
+++ b/sound/soc/fsl/fsl_spdif.c
@@ -1023,6 +1023,15 @@ static bool fsl_spdif_volatile_reg(struct device *dev, unsigned int reg)
 {
 	switch (reg) {
 	case REG_SPDIF_SIS:
+	case REG_SPDIF_SRL:
+	case REG_SPDIF_SRR:
+	case REG_SPDIF_SRCSH:
+	case REG_SPDIF_SRCSL:
+	case REG_SPDIF_SRU:
+	case REG_SPDIF_SRQ:
+	case REG_SPDIF_STL:
+	case REG_SPDIF_STR:
+	case REG_SPDIF_SRFM:
 		return true;
 	default:
 		return false;
-- 
1.9.1

